src/Type/Definition/ListOfType.php

```diff
@@ -4,24 +4,38 @@
 
 namespace GraphQL\Type\Definition;
 
+use GraphQL\Type\Schema;
+use function is_callable;
+
 class ListOfType extends Type implements WrappingType, OutputType, NullableType, InputType
 {
     /** @var Type */
     public $ofType;
 
-    public function __construct(Type $type)
+    /**
+     * @param callable():Type|Type $type
+     */
+    public function __construct($type)
     {
-        $this->ofType = $type;
+        $this->ofType = is_callable($type) ? $type : Type::assertType($type);
```

Comment:
```suggestion
        $this->ofType = is_callable($type)
            ? $type
            : Type::assertType($type);
```

This is a bit of a pet peeve of mine, i find one-liner ternary expressions wildly unreadable.

Reply:
If you feel that strongly about it then it should be enforced by the linter. Since it's not, and since I personally prefer the one-liner, I'd rather not descend into subjective style nits, if it's all the same to you.

Reply:
FWIW I'm with @spawnia 

Too easy to miss the else part when reading the code.

Reply:
I'm not going to get into some popularity vote over this. You are free to open your own PR after the fact and do whatever kind of style gymnastics you want and I wont' interfere, but I have enough trouble with this PR without bikeshedding.

Reply:
`php-cs-fixer` doesn't seem to have a configuration rule for it. Would be nice to automate for sure.